ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveSubQueryRemoveRule.java

```diff
@@ -406,6 +409,16 @@ private RexNode rewriteInExists(RexSubQuery e, Set<CorrelationId> variablesSet,
         offset = offset + 1;
         builder.push(e.rel);
       }
+    } else if (e.getKind() == SqlKind.EXISTS && !variablesSet.isEmpty()) {
+      // Query has 'exists' and correlation:
```

Comment:
Can you add a comment to explain/mention why limit is being removed for correlated EXISTs?

Reply:
Added comment